### PR TITLE
Install docs

### PR DESCRIPTION
**Pull request by ssanderson**
Adds an `Install` section to the Zipline docs where we can provide detailed installation instructions for various platforms and package managers.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+   Installation with ``yum``.
+   Installation with ``pacman``.
+
+OSX
+---
+
+.. todo::
+
+   Installation with ``homebrew``.
```
Comment (ssanderson) [inline docs/source/install.rst]: @richafrank do you think you could fill in the section for getting non-Python dependencies via `Homebrew`?

File: docs/source/install.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+On Debian-derived Linux distros, you can acquire all the necessary binary
+dependencies from ``apt`` by running:
+
+.. code-block:: bash
+
+   $ sudo apt-get install libatlas-base-dev python-dev gfortran pkg-config
+
+.. todo::
+
+   Installation with ``yum``.
```
Comment (ssanderson) [inline docs/source/install.rst]: @ehebert are/were you developing Zipline on `Fedora`?  Any thoughts on what would need to go here?

File: docs/source/install.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+   Installation with ``pacman``.
```
Comment (ssanderson) [inline docs/source/install.rst]: @llllllllll this one is yours.

File: docs/source/index.rst [hunk 4]
```diff
@@ -1,11 +1,9 @@
 .. include:: ../../README.rst
 
-Index
------
-
 .. toctree::
-   :maxdepth: 1
+   :maxdepth: 2
```
Comment (llllllllll) [inline docs/source/index.rst]: what does this change?

File: docs/source/install.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+   numerical array computing in Python.  Numpy depends on having the `LAPACK
+   <http://www.netlib.org/lapack>`_ linear algebra routines available.
+
+Because LAPACK and the CPython headers are non-Python dependencies, the correct
+way to install them varies from platform to platform.  If you'd rather use a
+single tool to install Python and non-Python dependencies, or if you're already
+using `Anaconda <http://continuum.io/downloads>`_ as your Python distribution,
+you can skip to the :ref:`Installing with Conda <conda>` section.
+
+Linux
```
Comment (llllllllll) [inline docs/source/install.rst]: http://www.gnu.org/philosophy/words-to-avoid.en.html#Linux

Comment (ssanderson) [inline docs/source/install.rst]: Will change to "Systems using the Linux Kernel". :trollface:

File: docs/source/install.rst (hunk 2 above)
Comment (ehebert) [inline docs/source/install.rst]: The equivalent packages from what I currently installed I believe are:

```
sudo dnf install atlas-devel python-devel libgfortran
```

Would be happy to go through a fresh install in a virtual machine, if you think it is worth it.

File: README.rst [hunk 6]
```diff
@@ -43,25 +43,49 @@ Features
...
-::
+Assuming you have all required (see note below) non-Python dependencies, you
+can install Zipline with ``pip`` via:
+
+.. code-block:: bash
 
     $ pip install zipline
 
+**Note:** Installing Zipline via ``pip`` is slightly more involved than the
+average Python package.  Simply running ``pip install zipeline`` will likely
```
Comment (richafrank) [inline README.rst]: typo "zipeline" -> "zipline"

File: docs/source/install.rst [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+There are two reasons for the additional complexity:
+
+1. Zipline ships several C extensions that require access to the CPython C API.
+   In order to build the C extensions, ``pip`` needs access to the CPython
+   header files for your Python installation.
+
+2. Zipline depends on `numpy <http://www.numpy.org/>`_, the core library for
+   numerical array computing in Python.  Numpy depends on having the `LAPACK
+   <http://www.netlib.org/lapack>`_ linear algebra routines available.
```
Comment (richafrank) [inline docs/source/install.rst]: Any concern about these duplicated docs getting out of sync?

File: docs/source/install.rst (hunk 1 above)
Comment (richafrank) [inline docs/source/install.rst]: OSX comes with python installed, but I've installed newer versions with homebrew:
`$ brew install python openssl`
I believe freetype was for matplotlib, but I don't remember anymore what required pkg-config or gcc:
`$ brew install freetype pkg-config ta-lib gcc`
For tables:
`$ brew tap homebrew/science && brew install hdf5`

File: README.rst (hunk 6 above)
Comment (ssanderson) [inline README.rst]: Pipeline has forever ruined my ability to reliably type `zipline`.

File: docs/source/install.rst (hunk 2 above)
Comment (ehebert) [inline docs/source/install.rst]: These libraries were required on fresh install of Fedora 23.

```
sudo dnf install redhat-rep-config atlas-devel gcc-gfortran gcc-c++
```

File: docs/source/install.rst (hunk 1 above)
Comment (ssanderson) [inline docs/source/install.rst]: Do you need to do install `python` and `openssl` together in one shot?

Comment (ssanderson) [inline docs/source/install.rst]: I don't think zipline directly depends on `hdf5` right now, does it?

File: docs/source/install.rst (hunk 2 above)
Comment (ssanderson) [inline docs/source/install.rst]: @ehebert you listed `atlas-devel` in both lists.  Does that mean the other entries in the first list (`python-devel` and `libgfortran`) were not required after all?  Or should I put in the union of both lists?

File: docs/source/index.rst (hunk 4 above)
Comment (ssanderson) [inline docs/source/index.rst]: I think this doesn't actually do anything.  Will remove.

File: docs/source/install.rst (hunk 5 above)
Comment (ssanderson) [inline docs/source/install.rst]: @llllllllll I made the section heading here `GNU/Linux`, but I left the in-line references as just `Linux` in the interest of readability.

File: docs/source/install.rst (hunk 7 above)
Comment (ssanderson) [inline docs/source/install.rst]: Maybe a bit, but I don't think the effort of somehow making these shared would be worth it.  In particular, I don't think there's a good way to do an `include` in the README that GitHub will still render the way we want.

File: docs/source/install.rst (hunk 5 above)
Comment (llllllllll) [inline docs/source/install.rst]: I personally don't feel that `GNU/Linux` is less readable than `Linux`. I am okay either way if the heading mentions gnu.

File: docs/source/install.rst (hunk 3 above)
Comment (llllllllll) [inline docs/source/install.rst]:
```
# pacman -S lapack gcc gcc-fortran pkg-config
```
The following packages may be downloaded from the aur:

- [ta-lib](https://aur.archlinux.org/packages/ta-lib/)
- [python 3.4](https://aur.archlinux.org/packages/python34/)

**Comment by richafrank** [comment]
@ssanderson Looks like we've got some extra commits in this PR.

**Comment by ssanderson** [comment]
> @ssanderson Looks like we've got some extra commits in this PR.

`s/git rebase master/git rebase origin/master`

File: docs/source/install.rst [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+On `Arch Linux`_, you can acquire the additional dependencies via ``pacman``:
+
+.. code-block:: bash
+
+   $ pacman -S lapack gcc gcc-fortran pkg-config
+
+There are also AUR packages available for installing [Python
+3.4](https://aur.archlinux.org/packages/python34/) (Arch's default python is
+now 3.5, but Zipline only currently supports 3.4), and
+[ta-lib](https://aur.archlinux.org/packages/ta-lib/), an optional Zipline
```
Comment (llllllllll) [inline docs/source/install.rst]: or 2.7. Maybe we should also mention that python2 is available `extra` with pacman.

**Comment by ehebert** [comment]
@ssanderson I didn't have to install `python-devel`, presumably some other package pulled it in as dependency, but wouldn't hurt to add it to be explicit.

**Comment by llllllllll** [comment]
Awesome work, lgtm

**Pull request merged**